falcon/hooks.py

```diff
@@ -89,7 +107,7 @@ def _before(responder_or_resource):
                     # responder in the do_before_all closure; otherwise, they
                     # will capture the same responder variable that is shared
                     # between iterations of the for loop, above.
-                    def let(responder=responder):
+                    def let(responder: Resource = responder) -> None:
```

Comment:
This should use a more specific type that is yet undefined, `Responder`.

Reply:
Changed.